8.17.10 release notes (#7021)
elastic/security-docs: docs/release-notes/8.17.asciidoc

[[release-notes-header-8.17.0]]
== 8.17

[discrete]
[[release-notes-8.17.10]]
=== 8.17.10

[discrete]
[[enhancements-8.17.10]]
==== Enhancements
* Due to an issue in macOS, {elastic-defend} would sometimes send network events without `user.name` populated. {elastic-defend} now identifies these events and populates `user.name` if necessary.

> * Reduces {elastic-defend} CPU when processing events from the System process.

Reviewer: ```suggestion
* Reduces {elastic-defend} CPU when processing events from the System process.
* Reduces {elastic-defend} CPU usage for ETW events, API events, and Behavioral Protections. In some cases, this may be a significant reduction.
```

[discrete]
[[bug-fixes-8.17.10]]
==== Fixes
* Fixes a race condition in {elastic-defend} on Windows that occasionally resulted in corrupted process command lines. This could cause incorrect values for `process.command_line`, `process.args_count`, and `process.args`, leading to false positives.

> * Improves the efficiency of the {elastic-defend} malware scan queue by not blocking scan requests when an oplock for the file being scanned cannot be acquired.

Reviewer: These two are still pending backports from @bit-envoy

Author: Hi @bit-envoy, have these two PRs made it into 8.17.10?

Reviewer: They built another BC of 8.17.10 with all the referenced Defend changes.

> * Fixes an issue in {elastic-defend} performance metrics that resulted in `endpoint_uptime_percent` always being 0 for behavioral rules.

Reviewer: ```suggestion
* Fixes an issue in {elastic-defend} performance metrics that resulted in `endpoint_uptime_percent` always being 0 for behavioral rules.
* Fixes an issue in {elastic-defend} that could result in a crash if a Logstash output configuration is specified containing a certificate which cannot not be parsed.
* Shortens the time it takes for {elastic-defend} to recover from a DEGRADED status caused by Elastic Agent communication issues.
```

Author: Hey @gabriellandau – we had "Shortens the time it takes for {elastic-defend} to recover from a DEGRADED status caused by Elastic Agent communication issues."  documented in the 8.17.9 release notes – should we remove it from there?

Reviewer: Sorry that PR was mislabeled.  It was probably intended for 8.17.9, but it was merged into the 8.17 branch after the 8.17.9 BC was built.  I just fixed the label.

== 8.17

[discrete]
[[release-notes-8.17.10]]
=== 8.17.10

[discrete]
[[enhancements-8.17.10]]
==== Enhancements
* Due to an issue in macOS, {elastic-defend} would sometimes send network events without `user.name` populated. {elastic-defend} now identifies these events and populates `user.name` if necessary.

> * Reduces {elastic-defend} CPU when processing events from the System process.

Reviewer 2: Should it be CPU usage -- or just CPU?

* Reduces {elastic-defend} CPU usage for ETW events, API events, and Behavioral Protections. In some cases, this may be a significant reduction.

[discrete]
[[bug-fixes-8.17.10]]
==== Fixes
* Fixes a race condition in {elastic-defend} on Windows that occasionally resulted in corrupted process command lines. This could cause incorrect values for `process.command_line`, `process.args_count`, and `process.args`, leading to false positives.
* Improves the efficiency of the {elastic-defend} malware scan queue by not blocking scan requests when an oplock for the file being scanned cannot be acquired.
* Fixes an issue in {elastic-defend} performance metrics that resulted in `endpoint_uptime_percent` always being 0 for behavioral rules.

> * Fixes an issue in {elastic-defend} that could result in a crash if a {ls} output configuration is specified containing a certificate which cannot not be parsed.

Reviewer 2: This sentence sounds a little awkward? Alt suggestion but not sure if it still means the same. 
```suggestion
* Fixes an issue in {elastic-defend} that could result in a crash if a {ls} output configuration contains a specified certificate that cannot not be parsed.
```
